lib/Echidna/Mutator.hs

```diff
@@ -0,0 +1,86 @@
+module Echidna.Mutator where
+
+import Control.Monad.Random.Strict (fromList, MonadRandom, getRandomR)
+import Data.Maybe (maybe)
+import Data.ByteString (ByteString, pack, unpack)
+import GHC.Word (Word8)
+import EVM.ABI (AbiValue(..))
+
+listMutators :: MonadRandom m => m ([a] -> m [a])
+listMutators = fromList [(return, 1), (expandRandList, 10), (deleteRandList, 10), (swapRandList, 10)] 
+ 
+mutateV :: MonadRandom m => Maybe Int -> [AbiValue] -> [AbiValue] -> m [AbiValue]
+mutateV mn fs vs = do
+  f <- listMutators
+  xs <- f vs
+  return $ maybe xs (`take` (xs ++ fs)) mn
+
+mutateBS :: MonadRandom m => Maybe Int -> [Word8] -> ByteString -> m ByteString
+mutateBS mn fs bs = do
+  f <- listMutators
+  xs <- f $ unpack bs
+  return $ pack $ maybe xs (`take` (xs ++ fs)) mn
+
+
+replaceAt :: a -> [a] -> Int -> [a]
+replaceAt _ []     _ = []
+replaceAt a (_:xs) 0 = a:xs
+replaceAt a (x:xs) n =
+  if n < 0
+  then x : xs
+  else x : replaceAt a xs (n-1)
+
+
+expandRandList :: MonadRandom m => [a] -> m [a]
+expandRandList []   = return []
+expandRandList xs = do k <- getRandomR (0, length xs - 1)
+                       t <- getRandomR (1, max 32 (length xs))
+                       return $ insertAt (xs !! k) xs t
```

Comment:
you can use `uniform xs` instead

Reply:
This was modified to requiere only using the index (`k`). I could use `uniform [0 .. length xs - 1]`, which is certainly more elegant that `getRandomR (0, length xs - 1)`, however, I believe using `getRandomR` will be faster at the cost of some code slightly less easier to read.